data/pools/transactionPool.go

```diff
@@ -233,6 +233,16 @@ func (pool *TransactionPool) test(txgroup []transactions.SignedTxn) error {
 	// requires a flat MinTxnFee).
 	feePerByte = feePerByte * pool.feeThresholdMultiplier
 
+	// The threshold grows exponentially if there are multiple blocks
+	// pending in the pool.
+	if pool.numPendingWholeBlocks > 1 {
+		// golang has no convenient integer exponentiation, so we just
+		// do this in a loop
+		for i := 0; i < int(pool.numPendingWholeBlocks)-1; i++ {
+			feePerByte *= pool.expFeeFactor
+		}
+	}
```

Comment:
nit : I think it would be better to have the multiplier being calculated along with `pool.numPendingWholeBlocks`, so that we could always multiply the feePerByte with that multiplier.
Not only that it would be faster, but it would also have constant time.

btw - Correct me if I'm wrong, but with this change, you're practically rolling back your previous change while making it `pool.expFeeFactor` adjustable, right ?

Reply:
That would not be the case. Since if `pool.expFeeFactor` set to 1, then there is still no fee increase.